### PR TITLE
Fix incorrect optional parameters

### PR DESCRIPTION
**Pull request by lquanx**
Issue #, if available:

Description of changes:

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by lquanx** [comment]
Previously the optional parameters would be wrapped by single quote which would cause errors.
image

**Comment by lquanx** [comment]
This PR would remove the single quote

**Review by lwarren01 — COMMENTED** [review]
I am confused why the old version didn't work. If the parameter was null or unset, then nothing within the expansion is substituted.

**Comment by lquanx** [comment]
patientPickerEndpoint was set. I would work, if this is not set

**Comment by nirojshrestha019** [comment]
we are only sending `patientPickerEndpoint` variable value through terraform aws fhir works correct? `issuerEndpoint` and `oAuth2ApiEndpoint` is not passed through our module currently so I was wondering if there values are empty or we will be passing some values from terraform module while calling install.sh? Sorry if this is stupid question.

**Comment by lwarren01** [comment]
I believe the issue is because the default values for these parameters are empty string. See line 185. I think if no value is set, then the previous version works.

**Comment by lwarren01** [comment]
https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html

${parameter:+word}

    If parameter is null or unset, nothing is substituted, otherwise the expansion of word is substituted.

**Comment by nirojshrestha019** [comment]
we are only setting value for `patientPickerEndpoint` variable if `smartAuthDeploymentEnabled`=TRUE in build spec `smartParams` variable. And we are passing this as parameter while calling install.sh

**Comment by lwarren01** [comment]
> we are only setting value for patientPickerEndpoint variable if smartAuthDeploymentEnabled=TRUE in build spec smartParams variable. And we are passing this as parameter while calling install.sh

Right. We need to keep that one for now for deployments into dev. But I am not following you. Did I miss something? Are the other parameter values not being set to empty string in this file?

**Comment by nirojshrestha019** [comment]
> > we are only setting value for patientPickerEndpoint variable if smartAuthDeploymentEnabled=TRUE in build spec smartParams variable. And we are passing this as parameter while calling install.sh
> 
> Right. We need to keep that one for now for deployments into dev. But I am not following you. Did I miss something? Are the other parameter values not being set to empty string in this file?

We are only passing `patientPickerEndpoint` variable value into install.sh call and by default like you mentioned before they are set to empty string. According to the install readme of FWOA (https://github.com/awslabs/fhir-works-on-aws-deployment/blob/smart-mainline/INSTALL.md) `issuerEndpoint` , `oAuth2ApiEndpoint`, `patientPickerEndpoint` all three parameters are required. Can you please confirm these parameter values for the sandbox account just to be sure: 
* --issuerEndpoint "https://id-sbx.resmed.com/oauth2/aus115doo99N6zKxt0h8" 
* --oAuth2ApiEndpoint "https://id-sbx.resmed.com/oauth2/aus115doo99N6zKxt0h8/v1/authorize" 
* --patientPickerEndpoint "https://uxuj41rmw1.execute-api.us-west-2.amazonaws.com/dev"

**Comment by lwarren01** [comment]
I am pretty certain that if you comment out the following lines from install.sh, you achieve the same result you have provided here:
```
#issuerEndpoint=""
#oAuth2ApiEndpoint=""
#patientPickerEndpoint=""
```
For example, this script will echo "--var <parameter>" if a parameter is provided to the script, else nothing is echoed.
```
#!/bin/sh
if [ "$1" != "" ]; then param=${1}; fi
echo ${param:+"--param $var"}
```

**Comment by lquanx** [comment]
> I am pretty certain that if you comment out the following lines from install.sh, you achieve the same result you have provided here:
> 
> ```
> #issuerEndpoint=""
> #oAuth2ApiEndpoint=""
> #patientPickerEndpoint=""
> ```
> 
> For example, this script will echo "--var " if a parameter is provided to the script, else nothing is echoed.
> 
> ```
> #!/bin/sh
> if [ "$1" != "" ]; then param=${1}; fi
> echo ${param:+"--param $var"}
> ```

Yes, I understand what you mean. I will make code changes.

File: scripts/install.sh [hunk 1]
```diff
@@ -182,9 +182,9 @@ then
 fi
 
 #Default values
-issuerEndpoint=""
-oAuth2ApiEndpoint=""
-patientPickerEndpoint=""
+#issuerEndpoint=""
+#oAuth2ApiEndpoint=""
+#patientPickerEndpoint=""
```
Comment (lquanx) [inline scripts/install.sh]: @lwarren01 I have commented out the default values.

Comment (lquanx) [reply scripts/install.sh]: But I don't think this would fix the issue. Lets talk about it over zoom

**Review by lwarren01 — APPROVED** [review]

**Pull request merged**